commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala

```diff
@@ -204,31 +210,31 @@ trait PurePursuitControllers extends UnicycleCoreControllers {
     }
 
     val (turnOutput, multiplier, lookAheadPoint) = purePursuitControllerTurn(turnPosition, position, selectedPath)
-    val historyUpdatingLookAheadPoint = lookAheadPoint.withCheck{p =>
+    val lookAheadHandle = lookAheadPoint.foreach{ p =>
       previousLookAheadPoint = Some(p)
     }
 
     val (forwardOutput, forwardError) = pointDistanceControl(
       position,
-      historyUpdatingLookAheadPoint)
-    val distanceToLast = position.map(_ distanceTo wayPoints.last)
+      selectedPath.map(_._1.end))
+    val distanceToLast = position.map{ pose =>
+      pose distanceTo wayPoints.last
+    }
 
     val errorToLast = distanceToLast.map { d =>
-      // error does not exist of we are not on our last segment
+      // error does not exist if we are not on our last segment
       if (!biSegmentPaths.hasNext) {
         Some(d)
       } else {
         None
       }
-//      (if (!biSegmentPaths.hasNext) {
-//        Some(d)
-//      } else {
-//        None
-//      }).flatten
     }
 
-    (forwardOutput.zip(turnOutput).zip(multiplier).zip(distanceToLast).zip(forwardError).map { o =>
-      val ((((forward, turn), fdMultiplier), _), frdError) = o
+    (forwardOutput.zip(turnOutput).zip(multiplier).zip(lookAheadPoint).zip(forwardError).map { o =>
+      val ((((forward, turn), fdMultiplier), la), frdError) = o
+
+      val hard = hardware
```

Comment:
Is there any reason you're creating this alias?

Reply:
This is for debug purposes, to be able to get details such as position, velocity, and info about the state of the robot when in debug mode